docs(marketplace): init section from Add-on provider API page (#303)
CleverCloud/documentation: content/doc/marketplace/_index.md

---
title: Marketplace APIs & Tools
shortdesc: How to integrate your own service as an add-on on Clever Cloud's Marketplace
keywords:
- Add-on
- API
- Marketplace
- Partners
aliases:
- /doc/extend/addon-api/
type: docs
comments: false
weight: 13

---

> Clever Cloud allows its Marketplace partners to provide services as add-ons with revenue sharing. Thus, they can be available to purchase and provision from the [Console](https://console.clever-cloud.com), [Clever Tools](https://github.com/CleverCloud/clever-tools), the [API](/api) or other integrations such as the [Terraform provider](https://registry.terraform.io/providers/CleverCloud/clevercloud/).

Reviewer: ```suggestion
{< hextra/hero-subtitle >}} Clever Cloud allows its Marketplace partners to provide services as add-ons with revenue sharing. Thus, they can be available to purchase and provision from the [Console](https://console.clever-cloud.com), [Clever Tools](https://github.com/CleverCloud/clever-tools), the [API](/api) or other integrations such as the [Terraform provider](https://registry.terraform.io/providers/CleverCloud/clevercloud/).{{< /hextra/hero-subtitle >}}

```

Author: I prefer not to rely on hextra specific templating when not necessary

Fields

* `id` - An ID for your add-on. All lower case, no spaces or punctuation. Underscores and dashes are allowed. This can’t be changed after the first push. It is also used for HTTP basic auth when making provisioning calls.

> * `name` (Optional) - An human-readable name for your add-on. You will be able to change it later in the dashboard, so you don't even have to provide it right now.

Reviewer: ```suggestion
* `name` (Optional) - A human-readable name for your add-on. You will be able to change it later in the dashboard, so you don't even have to provide it right now.
```

> * `api/config_vars` - A list of config vars that will be returned on provisioning calls. Each `config_var` name *MUST* start with the capitalized, add-on id with underscores, as in the example.

Reviewer: ```suggestion
* `api/config_vars` - A list of configuration variables that will be returned on provisioning calls. Each `config_var` name must start with the capitalized, add-on id with underscores, as in the example.
```

What does this sentence even mean? "Must be written in capital letters", maybe?

* `api/password` - Password that Clever Cloud will send in HTTP basic auth when making provisioning calls. You should generate a long random string for that field.

* `api/sso_salt` - Shared secret used in single sign-on between the Clever Cloud admin panel and your service’s admin panel. You should generate a long random string for that field.

* `api/regions` - The list of geographical zones supported by your add-on. It cannot be empty. As for now, it *MUST* contain the element "eu". More will be supported.

> * `api/production/base_url` - The production endpoint on which Clever Cloud will send actions requests (provision, deprovision, and plan change).

Reviewer: ```suggestion
* `api/production/base_url` - The production endpoint on which Clever Cloud sends actions requests (provision, deprovision, and plan change).
```

* `api/production/sso_url` - The production endpoint for single sign-on.

> * `api/test/base_url` - The test endpoint on which Clever Cloud will send actions requests. Used to test your service when you create an add-on provider. Then it is the `api/production/base_url` which is used.

Reviewer: ```suggestion
* `api/test/base_url` - The test endpoint on which Clever Cloud sends actions requests. Used to test your service when you create an add-on provider. After the add-on creation,`api/production/base_url` is used.
```

Unclear sentence. Maybe rework this later?

> * `api/test/sso_url` - The test endpoint for single sign-on. Used to test your service when you create an add-on provider. Then it is the `api/production/sso_url` which is used.

Reviewer: ```suggestion
* `api/test/sso_url` - The test endpoint for single sign-on. Used to test your service when you create an add-on provider. After that, the `api/production/sso_url` is used.
```

* `domains` - Originally the domains names for the application owning the add-on. We return an empty list.

> ### Update the config vars for an add-on

Reviewer: ```suggestion
### Update the configuration variables for an add-on
```
